### PR TITLE
Add a jooq ExecutionListener to the DSLContext so that we can wrap Da…

### PR DESCRIPTION
**Pull request by rma-rripken**
…taAccessException.  Also added a connection method that will catch and wrap exceptions.

### CODE REVIEW

**Comment by adamkorynta** [comment]
LOVE this

File: cwms_radar_api/src/main/java/cwms/radar/data/dao/JooqDao.java [hunk 1]
```diff
@@ -80,4 +94,89 @@ public void accept(org.jooq.Context<?> ctx) {
...
+            if (parts.length > 1) {
+                exception = new NotFoundException(parts[0]);
+            }
+        }
+        return exception;
+    }
+
+
+
+    private static class ExceptionWrappingListener extends DefaultExecuteListener {
```
Comment (adamkorynta) [inline cwms_radar_api/src/main/java/cwms/radar/data/dao/JooqDao.java]: styleguide nitpick: inner classes at the bottom of the file

**Review by adamkorynta — DISMISSED** [review]

File: cwms_radar_api/src/main/java/cwms/radar/data/dao/JooqDao.java (hunk 1 above)
Comment (rma-rripken) [reply cwms_radar_api/src/main/java/cwms/radar/data/dao/JooqDao.java]: I'd say this PR is currently only half-baked.  Hopefully discussion in #299 will lead to a more complete approach.

File: cwms_radar_api/src/main/java/cwms/radar/api/LevelsController.java [hunk 2]
```diff
@@ -356,6 +356,8 @@ String.class, null, metrics, name(LevelsController.class.getName(),
                     UnitSystem.EN.getValue(), unmarshalledDateTime, office);
             ctx.json(locationLevel);
             ctx.status(HttpServletResponse.SC_OK);
+        } catch (NotFoundException e){
```
Comment (MikeNeilson) [inline cwms_radar_api/src/main/java/cwms/radar/api/LevelsController.java:359]: Arguably we could ditch both of these and just always rethrow. Though if I'm remembering right try{} doesn't supress an exception.

**Review by MikeNeilson — APPROVED** [review]

**Pull request merged**